### PR TITLE
[move-2024][migration] Update `games` to use Move 2024 Syntax

### PR DESCRIPTION
**Pull request by cgswords**
## Description 

Updates the `games` examples directory to use Move 2024 syntax

## Test plan 

Local disassembly testing, plus all the tests still work. Would appreciate a careful review, though!

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by jonas-lj — APPROVED** [review]

File: sui_programmability/examples/games/sources/hero.move [hunk 1]
```diff
@@ -349,47 +349,48 @@ module games::hero {
     #[test]
     fun slay_boar_test() {
         use sui::test_scenario;
+        use sui::coin;
 
         let admin = @0xAD014;
         let player = @0x0;
 
         let mut scenario_val = test_scenario::begin(admin);
         let scenario = &mut scenario_val;
```
[the remark this answers is not in the document]
Comment (tnowacki) [reply sui_programmability/examples/games/sources/hero.move:358]: +1

File: sui_programmability/examples/games/tests/tic_tac_toe_tests.move [hunk 2]
```diff
@@ -19,7 +19,7 @@ module games::tic_tac_toe_tests {
         let mut scenario_val = test_scenario::begin(admin);
         let scenario = &mut scenario_val;
         // Admin creates a game
-        tic_tac_toe::create_game(copy player_x, copy player_o, test_scenario::ctx(scenario));
+        tic_tac_toe::create_game(copy player_x, copy player_o, scenario.ctx());
```
[the remark this answers is not in the document]
Comment (cgswords) [reply sui_programmability/examples/games/tests/tic_tac_toe_tests.move:22]: Agreed, and removing it appears to work fine!

File: sui_programmability/examples/games/sources/hero.move (hunk 1 above)
[the remark this answers is not in the document]
Comment (cgswords) [reply sui_programmability/examples/games/sources/hero.move:358]: This isn't always possible without significant rewrites. The tic tac toe, raffles, and slots tests, for example, would trade this binding for explicitly passing `&mut scenario` in a large number of places.

I've done it where it's practical.

File: sui_programmability/examples/games/sources/hero.move [hunk 3]
```diff
@@ -263,7 +263,7 @@ module games::hero {
     ) {
         let sword = create_sword(game, payment, ctx);
         let hero = create_hero(game, sword, ctx);
-        transfer::public_transfer(hero, tx_context::sender(ctx))
+        transfer::public_transfer(hero, ctx.sender())
     }
 
     /// Anyone can create a hero if they have a sword. All heroes start with the
```
[the remark this answers is not in the document]
Comment (cgswords) [reply sui_programmability/examples/games/sources/hero.move:269]: Yep, plus a few other places.

**Pull request merged**